frontend/src/views/Room.jsx

```diff
@@ -105,21 +109,60 @@ function Room() {
     }
   };
 
+  const handleRoleChange = (payload) => {
+    if (payload.eventType === 'INSERT') {
+      const { id, userEmail, permissionId } = payload.new;
+      dispatch(addUpdateParticipant({
+        name: userEmail,
+        role: permissionId === 1 ? ROLES.HOST : ROLES.PRESENTER,
+        id,
+      }));
+    }
+    // Supabase realtime only sends the ID that was deleted from the rooms-data table
+    if (payload.eventType === 'DELETE') {
+      const { id } = payload.old;
+      dispatch(removeRole({ id }));
+    }
+  };
+
   // initialize room
   useEffect(() => {
+    const updateParticipantRoles = async () => {
+      const initialParticipantRoles = await getRoomPermissions(roomId);
+      initialParticipantRoles.map((part) => dispatch(addUpdateParticipant({
+        name: part.userEmail,
+        role: part.permissionId === 1 ? ROLES.HOST : ROLES.PRESENTER,
+        id: part.id,
+      })));
+    };
     const subscribeToRemoteStreams = async (r) => {
       // subscribe ta all remote participants for testing purposes
       const { remoteParticipants } = r;
       const rps = Array.from(remoteParticipants.values());
       await Promise.all(rps.map(async (rp) => {
         await rp.subscribe();
+        // Add remote participants to participants list.
+        dispatch(addUpdateParticipant({
+          name: rp.id,
+          role: ROLES.GUEST,
```

Comment:
Our WebRTC wrapper includes a `role` field for participants. Should we just use that? Perhaps overwriting what Mux provides us, and just using our own.

Reply:
I don't think that would work with out current architecture: The role we pass to the participant is static, and room administrators / hosts can change participants roles while they are in the room. They can pass a guest to presenter, presenter to guest, etc.
That's why I'm subscribed to the room-data table, so that if roles change in the middle of a meeting, the FE gets updated accordingly. For now, we just show the participant's  role but that should be replaced with custom UI according to the participant's role.
If it's possible to get the role and update it dynamically on the participant, I think that would be ideal! But if this works for now, maybe we can open a new ticket to get that done after.

Reply:
I get your point and I totally agree with subscribing to room-data table!

But also, I think you've already done what I suggested in my comment, because you call `updateParticipantRoles()` just after `subscribeToRemoteStreams()` :)
I believe we can further simplify the code: in `subscribeToRemoteStreams()`, we first dispatch a state update will all participants temporarily set as guest. Then, just after that, we call `updateParticipantRoles()` and dispatch another state update that partly overwrites the previous one... Couldn't we just call `getRoomPermissions(roomId)` before we do `await Promise.all()` and dispatch the right data immediately?